### PR TITLE
feat: add switch component

### PR DESCRIPTION
**Pull request by matthieuh**
@paulocfjunior on this PR I only enable tests on the Switch component since your are fixing the others in #80

### CODE REVIEW

File: packages/faency/src/components/Switch.story.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    <>
+      <Box mb="4">
+        <Text as="p">Uncontrolled</Text>
+        <Switch value="row" defaultChecked mr={1} />
+      </Box>
+
+      <Box mb="4">
+        <Text as="p">Uncontrolled and without children</Text>
+        <Switch value="true" defaultChecked />
+      </Box>
```
Comment (paulocfjunior) [inline packages/faency/src/components/Switch.story.tsx]: It seems to have no difference between the first and the second case. What is the difference? And what the `value="row"` stand for?

File: packages/faency/src/components/Switch.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      <FakeSwitch />
+    </SwitchWrapper>
+  )
+}
+
+const SwitchWrapper = styled('label')<MarginProps>(
+  css({
+    position: 'relative',
+    display: 'inline-block',
+    width: 6,
```
Comment (paulocfjunior) [inline packages/faency/src/components/Switch.tsx]: I tried with `width: 5` (and used `space.2` in the Transform for the `:after` pseudo-component) and it seems to have a better look and feel, what do you think?

Width: 6  (Original)
image

Width: 5 (Suggestion)
image

Comment (matthieuh) [reply packages/faency/src/components/Switch.tsx]: Indeed I'm agree it looks better! :)

File: packages/faency/src/components/Switch.story.tsx (hunk 1 above)
Comment (matthieuh) [reply packages/faency/src/components/Switch.story.tsx]: @paulocfjunior Fixed the unnecessary example. "row" is the value of the input. It's working the exact same way as an checkbox native element: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox

Comment (paulocfjunior) [reply packages/faency/src/components/Switch.story.tsx]: Got it, thanks for the link, that was very clarifying

**Comment by paulocfjunior** [comment]
This issue closes #74.

**Pull request merged**